src/motor_control/src/motor_control_node.cpp

```diff
@@ -188,6 +188,10 @@ class MotorControlNode : public rclcpp::Node {
 
     send_can(id + 0x00000300, data); // ID must be modified to signify this is a RPM command
     this->current_msg[id] = std::make_tuple(id + 0x00000300, data); // update the hashmap
+    while( rpm != this->can_data[id].velocity)
+    {
+      continue;

```

Comment:
Once again waiting in while loops like this is quite cpu intensive, try something event driven